decimal.go

```diff
@@ -1220,6 +1223,49 @@ func trailingZerosU128(n u128) uint8 {
 	return zeros
 }
 
+// PowToIntPart raises the decimal d to the power of integer part of e (d^int(e)).
+// This is useful when the exponent is an integer but stored in Decimal.
+//
+// Returns error if:
+//   - d is zero and e is a negative integer.
+//   - |int(e)| > math.MaxInt32 (because MaxInt32 is already ~2 billion, supporting more than that value is not practical and unnecessary).
```

Comment:
```suggestion
//   - |int(e)| > [math.MaxInt32] (because MaxInt32 is already ~2 billion, supporting more than that value is not practical and unnecessary).
```